### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by Seaear**
Add Docker support and Change *.zh-cn.resx to *.zh.resx (Msbuild does not support language-specific issues in Linux/Docker)

### CODE REVIEW

**Comment by skoruba** [comment]
@Seaear - Great work, I will check it.
Thanks!

**Comment by Seaear** [comment]
@skoruba 
Add missing Chinese translation

File: src/Skoruba.IdentityServer4.Admin/appsettings.Development.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿{
```
[the remark this answers is not in the document]
Comment (Seaear) [reply src/Skoruba.IdentityServer4.Admin/appsettings.Development.json]: This is not required, adding this file will only override default configuration in Docker.

**Comment by Seaear** [comment]
@skoruba Fix carsh on SQL Server 2008 R2

**Comment by xmichaelx** [comment]
Could you provide some instructions on how to build this?

File: docker-compose.override.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+version: '3.4'
+
+services:
+  skoruba.identityserver4.admin:
+    environment:
+      - ASPNETCORE_ENVIRONMENT=Development
```
Comment (xmichaelx) [inline docker-compose.override.yml]: If we really want to have different environmnet in Docker then let's have `Docker` environment instead of reusing Development env which is for development :)

This would also require to change `appsettings.Development.json` to `appsettings.Docker.json`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+version: '3.4'
+
+services:
+  skoruba.identityserver4.admin:
+    image: ${DOCKER_REGISTRY-}skoruba-identityserver4-admin
```
Comment (xmichaelx) [inline docker-compose.yml]: This ` ${DOCKER_REGISTRY-}` where is this variable set? When building on Linux this threw some errors.

File: src/Skoruba.IdentityServer4.Admin/Helpers/StartupHelpers.cs [hunk 4]
```diff
@@ -44,7 +44,7 @@ public static void RegisterDbContexts<TContext>(this IServiceCollection services
             var storeOptions = new ConfigurationStoreOptions();
             services.AddSingleton(storeOptions);
 
-            services.AddDbContext<TContext>(options => options.UseSqlServer(configuration.GetConnectionString(ConfigurationConsts.AdminConnectionStringKey), optionsSql => optionsSql.MigrationsAssembly(migrationsAssembly)));
+            services.AddDbContext<TContext>(options => options.UseSqlServer(configuration.GetConnectionString(ConfigurationConsts.AdminConnectionStringKey), optionsSql => optionsSql.MigrationsAssembly(migrationsAssembly).UseRowNumberForPaging()));
```
Comment (xmichaelx) [inline src/Skoruba.IdentityServer4.Admin/Helpers/StartupHelpers.cs]: This change should be a part of separate pull request since it has nothing to do with Docker.

File: src/Skoruba.IdentityServer4.Admin/Skoruba.IdentityServer4.Admin.csproj [hunk 5]
```diff
@@ -13,7 +15,8 @@
 		<PackageReference Include="IdentityServer4.EntityFramework" Version="2.3.2" />
 		<PackageReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="2.0.3" />
 		<PackageReference Include="Microsoft.Extensions.Options" Version="2.2.0" />
-		<PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.2.0" />
+    <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.0.2105168" />
```
Comment (xmichaelx) [inline src/Skoruba.IdentityServer4.Admin/Skoruba.IdentityServer4.Admin.csproj]: What is this package used for? @Seaear

File: src/Skoruba.IdentityServer4.Admin/Helpers/StartupHelpers.cs (hunk 4 above)
Comment (Seaear) [reply src/Skoruba.IdentityServer4.Admin/Helpers/StartupHelpers.cs]: I don't quite use GitHub.

File: docker-compose.yml (hunk 3 above)
Comment (Seaear) [reply docker-compose.yml]: I am using Visual Studio 2017, I have not tested in VSCode.

File: src/Skoruba.IdentityServer4.Admin/Skoruba.IdentityServer4.Admin.csproj (hunk 5 above)
Comment (Seaear) [reply src/Skoruba.IdentityServer4.Admin/Skoruba.IdentityServer4.Admin.csproj]: This package is required for VS 2017 to build Docker

File: docker-compose.override.yml (hunk 2 above)
Comment (Seaear) [reply docker-compose.override.yml]: In the future, I will adjust the database connection string to docker-compose.yml, appsettings.Development.json is not needed.

Comment (xmichaelx) [reply docker-compose.override.yml]: Yeah since @skoruba added `AddEnvironmentVariables` to configuration you can nicely set it up in environment in docker compose :) I agree that this is superior approach.

File: docker-compose.yml (hunk 3 above)
Comment (xmichaelx) [reply docker-compose.yml]: Could you provide couple of points or tutorial you followed to compose everything in VS?

File: src/Skoruba.IdentityServer4.Admin/Helpers/StartupHelpers.cs (hunk 4 above)
Comment (xmichaelx) [reply src/Skoruba.IdentityServer4.Admin/Helpers/StartupHelpers.cs]: It's not a problem - we will just skip this update in the PR and you can later open a new pull request.

Rule of thumb is to add to PR only stuff related to the title of pull request, we're very happy with you contributions - this is just protip for the future :)

File: docker-compose.yml (hunk 3 above)
Comment (Seaear) [reply docker-compose.yml]: https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/docker/visual-studio-tools-for-docker?view=aspnetcore-2.2

**Comment by Seaear** [comment]
@xmichaelx 
1. Add "127.0.0.1 id.example.com" to hosts file.
2. In the directory where docker-compose.yml, Use the command "docker-compose build" build services, and use "docker-compose up" start services.

**Comment by xmichaelx** [comment]
@Seaear ok i followed your tutorial and it works, so that's great. But I see that in compose file you have `admin.example.com` and `db.example.com` as aliases for for  sqlserver and admin container, should those be added to hosts file as well?

**Comment by Seaear** [comment]
@xmichaelx not needed for now.

**Comment by duki994** [comment]
> @skoruba Fix carsh on SQL Server 2008 R2

Using ROW_NUMBER command/query works down to SQL Server 2005 :)

**Comment by Jaben** [comment]
Why did you close this PR, @Seaear ? Is docker support not getting added?

**Comment by Seaear** [comment]
New PR https://github.com/skoruba/IdentityServer4.Admin/pull/178
@Jaben
